src/configuration/services/postgresql.md

```diff
@@ -71,6 +76,18 @@ foreach ($relationships['database'] as $endpoint) {
 }
 ```
 
+## Upgrading
+
+PostgreSQL does not support direct migration from one significant version to another without extensive user intervention. For that reason we do not support transparent updates from one PostgreSQL version to another (such as from 9.3 to 9.6).  In order to upgrade the service version, we recommend the following process:
+
+* Create a new service for the new PostgreSQL version
+* If possible, set your site to maintenance mode or read-only mode to avoid further database writes
+* Using `pg_dumpall` to take a snapshot of the old data
+* Import that data into the new service
+* Modify the relationships block in `.platform.app.yaml` to use the new service name and redeploy
```

Comment:
Unless I'm mistaken, you can't access a service at all without having a relationship to it (you connect via an app). So you'd need to have [also] modified your `relationships` block at the same time as "Create a new service".